langroid/agent/chat_agent.py

```diff
@@ -647,6 +674,12 @@ def enable_message(
                         set `_allow_llm_use=True` when you define the tool.
                         """
                     )
+                # Track whether `output_format` use was explicitly
+                # enabled and should not be disabled when
+                # `output_format` changes
+                if self.enabled_use_output_format is not None:
+                    if self.enabled_use_output_format.default_value("request") == t:
+                        self.enabled_use_output_format = None
```

Comment:
comment inside if block, right above this None assignment:

t was designated as "enabled for use" ONLY for output_format enforcement, but we are explicitly enabling it for use here, so we set the variable to None.